dotCMS/src/main/java/com/dotcms/webdav/FsFileResource.java

```diff
@@ -0,0 +1,133 @@
+import com.dotcms.repackage.com.bradmcevoy.io.ReadingException;
+import com.dotcms.repackage.com.bradmcevoy.io.WritingException;
+import com.dotcms.webdav.FileContentService;
+
+
+import java.io.*;
+import java.util.Map;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ */
+public class FsFileResource extends FsResource implements CopyableResource, DeletableResource, GetableResource, MoveableResource, PropFindableResource, PropPatchableResource {
+
+    private static final Logger log = LoggerFactory.getLogger(FsFileResource.class);
+    
+    private final FileContentService contentService;
+
+    /**
+     *
+     * @param host - the requested host. E.g. www.mycompany.com
+     * @param factory
+     * @param file
+     */
+    public FsFileResource(String host, FileSystemResourceFactory factory, File file, FileContentService contentService) {
+        super(host, factory, file);
+        this.contentService = contentService;
+    }
+
+    @Override
+    public Long getContentLength() {
+        return file.length();
+    }
+
+    @Override
+    public String getContentType(String preferredList) {
+        String mime = ContentTypeUtils.findContentTypes(this.file);
+        String s = ContentTypeUtils.findAcceptableContentType(mime, preferredList);
+        if (log.isTraceEnabled()) {
+            log.trace("getContentType: preferred: {} mime: {} selected: {}", new Object[]{preferredList, mime, s});
+        }
+        return s;
+    }
+
+    @Override
+    public String checkRedirect(Request arg0) {
+        return null;
+    }
+
+    @Override
+    public void sendContent(OutputStream out, Range range, Map<String, String> params, String contentType) throws IOException, NotFoundException {
+      try (InputStream in  = contentService.getFileContent(file)){
+            if (range != null) {
+                log.debug("sendContent: ranged content: " + file.getAbsolutePath());
+                PartialEntity.writeRange(in, range, out);
+            } else {
+                log.debug("sendContent: send whole file " + file.getAbsolutePath());
+                IOUtils.copy(in, out);
+            }
+            out.flush();
+        } catch (FileNotFoundException e) {
+            throw new NotFoundException("Couldnt locate content");
+        } catch (ReadingException e) {
+            throw new IOException(e);
+        } catch (WritingException e) {
+            throw new IOException(e);
+        } 
+    }
+
+    /**
+     * @{@inheritDoc}
+     */
+    @Override
+    public Long getMaxAgeSeconds(Auth auth) {
+        return factory.maxAgeSeconds(this);
+    }
+
+    /**
+     * @{@inheritDoc}
+     */
+    @Override
+    protected void doCopy(File dest) {
+        try {
+            FileUtils.copyFile(file, dest);
+        } catch (IOException ex) {
+            throw new RuntimeException("Failed doing copy to: " + dest.getAbsolutePath(), ex);

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)